internal/http/authenticator.go

```diff
@@ -0,0 +1,120 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/browser"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type UserAuthenticator struct {
+	conf *oauth2.Config
+}
+
+func NewUserAuthenticator(clientID, idpURL string) UserAuthenticator {
+	conf := &oauth2.Config{
+		ClientID: clientID,
+		Endpoint: oauth2.Endpoint{
+			TokenURL:      fmt.Sprintf("%s/v1/token", idpURL),
+			DeviceAuthURL: fmt.Sprintf("%s/v1/device/authorize", idpURL),
+		},
+		Scopes: []string{"openid profile"},
+	}
+	return UserAuthenticator{
+		conf: conf,
+	}
+}
+
+func (g *UserAuthenticator) Login() error {
+	ctx := context.Background()
```

Comment:
Maybe we should accept the context as an argument so the cobra CLI can cancel it correctly?